python_modules/libraries/dagster-dg/dagster_dg/cache.py

```diff
@@ -0,0 +1,66 @@
+import shutil
+import sys
+from pathlib import Path
+from typing import Final, Optional, Tuple
+
+from typing_extensions import Self
+
+from dagster_dg.config import DgConfig
+
+# All data is stored inside an additional container directory inside the user-specified cache
+# directory. When we clear the cache, we only delete this container directory. This is to avoid
+# accidents when the user mistakenly specifies a cache directory that contains other data.
+_CACHE_CONTAINER_NAME: Final = "dg-cache"
+
+
+def get_default_cache_dir() -> Path:
+    if sys.platform == "win32":
+        return Path.home() / "AppData" / "dg" / "cache"
+    elif sys.platform == "darwin":
+        return Path.home() / "Library" / "Caches" / "dg"
+    else:
+        return Path.home() / ".cache" / "dg"
+
+
+class DgCache:
+    @classmethod
+    def from_default(cls) -> Self:
+        return cls(get_default_cache_dir())
+
+    @classmethod
+    def from_config(cls, config: DgConfig) -> "DgCache":
+        return cls(
+            root_path=config.cache_dir,
+            logging_enabled=config.verbose,
+        )
+
+    def __init__(self, root_path: Path, logging_enabled: bool = False):
+        self._root_path = root_path / _CACHE_CONTAINER_NAME
```

Comment:
In general I think this sort of path manipuation should be done outside of a constructor to ease testability and maximize flexibility

Reply:
Good idea, moved to a `@classmethod`